qiskit_experiments/analysis/curve_analysis.py

```diff
@@ -0,0 +1,708 @@
+            )
+
+            # Generate fit options
+            fit_candidates = self._setup_fitting(_xdata, _ydata, _sigma, _series, **options)
+            if isinstance(fit_candidates, dict):
+                # only single initial guess
+                fit_options = self._format_fit_options(**fit_candidates)
+                fit_result = base_fitter(
+                    funcs=[series_def.fit_func for series_def in self.__series__],
+                    series=_series,
+                    xdata=_xdata,
+                    ydata=_ydata,
+                    sigma=_sigma,
+                    **fit_options,
+                )
+                analysis_result.update(**fit_result)
+            else:
+                # multiple initial guesses
+                fit_options_candidates = [
+                    self._format_fit_options(**fit_options) for fit_options in fit_candidates
+                ]
+                fit_results = [
+                    base_fitter(
+                        funcs=[series_def.fit_func for series_def in self.__series__],
+                        series=_series,
+                        xdata=_xdata,
+                        ydata=_ydata,
+                        sigma=_sigma,
+                        **fit_options,
+                    )
+                    for fit_options in fit_options_candidates
+                ]
+                # Sort by chi squared value
+                fit_results = sorted(fit_results, key=lambda r: r["reduced_chisq"])
+                analysis_result.update(**fit_results[0])
+            analysis_result["success"] = True
+        except Exception as ex:
+            analysis_result["error_message"] = str(ex)
+            analysis_result["success"] = False
+
+        #
+        # 4. Post-process analysis data
+        #
+        if analysis_result["success"]:
+            analysis_result = self._post_processing(analysis_result=analysis_result, **options)
+
+        #
+        # 5. Create figures
+        #
+        if plot:
+            figures = self._create_figures(
+                x_values=xdata,
+                y_values=ydata,
+                y_sigmas=sigma,
+                series=series,
+                analysis_results=analysis_result,
+                axis=axis,
+                xlabel=xlabel,
+                ylabel=ylabel,
+                fit_reports=fit_reports,
+            )
+        else:
+            figures = list()
+
+        #
+        # 6. Save raw data
+        #
+        raw_data_dict = dict()
+        for series_def in self.__series__:
+            sub_xdata, sub_ydata, sub_sigma = self._subset_data(
+                name=series_def.name, x_values=xdata, y_values=ydata, y_sigmas=sigma, series=series
+            )
+            raw_data_dict[series_def.name] = {
+                "xdata": sub_xdata,
+                "ydata": sub_ydata,
+                "sigma": sub_sigma,
+            }
+        analysis_result["raw_data"] = raw_data_dict
```

Comment:
I don't think raw data needs to be saved be default. You could make it an analysis option to return the raw data with default value of False.